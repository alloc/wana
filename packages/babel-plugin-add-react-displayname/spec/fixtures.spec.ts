const babel = require('@babel/core')
const fs = require('fs')
const path = require('path')
const assert = require('assert')

const fixturesDir = path.join(__dirname, 'fixtures')
const pluginPath = path.join(
  __dirname,
  '../../babel-plugin-add-react-displayname'
)

describe('add-react-displayname transform', () => {
  fs.readdirSync(fixturesDir).forEach(fixture => {
    const name = path.basename(fixture, path.extname(fixture))
    it('transforms ' + name, () => {
      const actual = transformFile(path.join(fixturesDir, fixture))
      expect(actual).toMatchSnapshot()
    })
  })
})

function readFile(filename) {
  let file = fs.readFileSync(filename, 'utf8').trim()
  file = file.replace(/\r\n/g, '\n')
  return file
}

function transformFile(filename) {
  return babel.transformFileSync(filename, {
    presets: ['@babel/react', '@babel/typescript'],
    plugins: [
      ['@babel/proposal-class-properties', { loose: true }],
      [pluginPath, { callees: ['withThing'] }],
    ],
  }).code
}
