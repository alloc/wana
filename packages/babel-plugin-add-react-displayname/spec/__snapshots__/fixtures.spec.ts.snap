// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add-react-displayname transform transforms calleesOption 1`] = `
"const Test1 = withThing(arg);
export const Test2 = withThing(() => null);
Test2.displayName = \\"Test2\\";"
`;

exports[`add-react-displayname transform transforms createClass 1`] = `
"// not supported
export const Test1 = React.createClass({
  displayName: \\"Test1\\",

  render() {
    return React.createElement(\\"div\\", null);
  }

});"
`;

exports[`add-react-displayname transform transforms multipleWraps 1`] = `
"export const Test1 = React.memo(React.forwardRef((props, ref) => {
  return React.createElement(\\"div\\", null);
}));
Test1.displayName = \\"Test1\\";
const Test2 = React.memo(React.forwardRef((props, ref) => {
  return React.createElement(\\"div\\", null);
}));
Test2.displayName = \\"Test2\\";"
`;

exports[`add-react-displayname transform transforms nestedInFunction 1`] = `
"export const Test1 = () => {
  const Test2 = withAuto(() => {
    return React.createElement(\\"div\\", null);
  });
  Test2.displayName = \\"Test2\\";
  return React.createElement(Test2, null);
};"
`;

exports[`add-react-displayname transform transforms nestedInObject 1`] = `
"const Foo = {
  Bar: withAuto(() => {
    return React.createElement(\\"div\\", null);
  })
};"
`;

exports[`add-react-displayname transform transforms singleWrap 1`] = `
"export const Test1 = withAuto(() => {
  return React.createElement(\\"div\\", null);
});
Test1.displayName = \\"Test1\\";
const Test2 = withAuto(() => {
  return React.createElement(\\"div\\", null);
});
Test2.displayName = \\"Test2\\";"
`;
